atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidator.java

```diff
@@ -159,8 +178,9 @@ Set<CassandraServer> getNewHostsWithInconsistentTopologies(
         if (currentServersWithoutSoftFailures.isEmpty()) {
             ClusterTopologyResult topologyResultFromNewServers =
                     maybeGetConsistentClusterTopology(newServersWithoutSoftFailures);
+            Set<String> configuredServersSnapshot = configuredServers.get();
             Map<CassandraServer, HostIdResult> newServersFromConfig = EntryStream.of(newServersWithoutSoftFailures)
-                    .filterKeys(server -> newlyAddedHosts.get(server) == CassandraServerOrigin.CONFIG)
+                    .filterKeys(server -> configuredServersSnapshot.contains(server.cassandraHostName()))
```

Comment:
note for future: This isn't necessary with a move to absent marking _after_ validation, but we need this in sooner than we need that in!